### PR TITLE
KDSingleApplication: Do not exceed maximum socket name length

### PR DESCRIPTION
**Pull request by jonaski**
In cases where there is a long name given (usually the application name) or a long username, `QLocalServer::listen` fails because `sockaddr_un.sun_path` has a char array of 108 on Linux, and 104 on BSD/macOS. On macOS, the TMPDIR path is long, ie.: `/var/folders/zg/b242mhvd125344h5zrnpkzk80000gn/T/`, so 49 characters just for the temp path alone. To fix this, use the uid instead of the username if the total length of the socket name is expected to exceed the maximum length. If it still exceeds the maximum allowed length, ie.: the given name is too long, chop the socket name which cuts off the last part of the name.

I've tested this to work on both Linux and macOS, Windows should be unaffected by these changes.

Fixes https://github.com/strawberrymusicplayer/strawberry/issues/1603

### CODE REVIEW

**Comment by jonaski** [comment]
The macOS CI is broken because `macos-latest` is arm64 based now, while you install x86_64 based Qt. It needs to be changed to macos-13, or install arm64 based Qt.

**Comment by jonaski** [comment]
#46 fixes the macOS build

**Comment by jonaski** [comment]
@iamsergio I added a test now

**Comment by iamsergio** [comment]
thanks! seems to be failing though ? on windows Qt 5

**Comment by jonaski** [comment]
Ping?

**Comment by nicolasfella** [comment]
Thanks!

**Review by dangelog — COMMENTED** [review]
Not a complete review, just some comments.

File: src/kdsingleapplication_localsocket.cpp [hunk 1]
```diff
@@ -50,52 +53,85 @@ KDSingleApplicationLocalSocket::KDSingleApplicationLocalSocket(const QString &na
...
+        userName = pw ? QString::fromUtf8(pw->pw_name) : alternativeUserName;
     }
     if (options.testFlag(KDSingleApplication::Option::IncludeSessionInSocketName)) {
-        QString sessionId = qEnvironmentVariable("XDG_SESSION_ID");
-        if (!sessionId.isEmpty()) {
-            m_socketName += QStringLiteral("-");
-            m_socketName += sessionId;
-        }
+        sessionId = qEnvironmentVariable("XDG_SESSION_ID");
+    }
```
[the remark this answers is not in the document]
Comment (jonaski) [reply src/kdsingleapplication_localsocket.cpp:79]: Will fix

File: src/kdsingleapplication_localsocket.cpp [hunk 2]
```diff
@@ -30,6 +30,9 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <pwd.h>
+#if defined(Q_OS_LINUX)
```
[the remark this answers is not in the document]
Comment (jonaski) [reply src/kdsingleapplication_localsocket.cpp:33]: https://github.com/KDAB/KDSingleApplication/pull/53

**Pull request merged**